### ISSUE TITLE
Retrieving attributes via reflection

### ISSUE THREAD

**Issue by DrSloth** [c3lang/c3c#1301]
It would be a great feature to be able query attributes via reflection.

This could allow things like setting the names or formats of fields for certain serialization formats, grouping fields or giving information for other macros.

**Comment by lerno** [comment]
There are two things here, one being able to get things with a particular attribute, one is to actually support data for those attributes. What is desired?

**Comment by lerno** [comment]
I don't think that custom attributes should introduce runtime information though?

**Comment by DrSloth** [comment]
> I don't think that custom attributes should introduce runtime information though?

I think it shouldn't by default. If the information should be queriable at runtime the user has to create a macro that generates arrays or maps from the attributes.

**Comment by lerno** [comment]
With the `@tag` feature, do we have this now?

**Comment by lerno** [comment]
What would that feature be used for though? That isn't already covered by querying whether the tag is there.

**Comment by lerno** [comment]
@Dodzey ping

**Comment by lerno** [comment]
I'll close this and consider it completed.
